File: bootstrap/archived_runa_0.0.7.5+/v0.0.8.4.5/benchmarks/c/binarysearch.c
#include <stdlib.h>

int main() {
    int n = 10000;
    int searches = 1000;
    int *arr = malloc(n * sizeof(int));

    for (int i = 0; i < n; i++) {
        arr[i] = i;
    }

    int total = 0;
    for (int i = 0; i < searches; i++) {
        int target = i % n;
        int left = 0;
        int right = n - 1;

        while (left <= right) {
            int mid = (left + right) / 2;
            if (arr[mid] == target) {
                total += mid;
                break;
            } else if (arr[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
    }

    free(arr);
    return 0;
}

File: bootstrap/v0.0.8.4.5.2.5/benchmarks/rust/binarysearch.rs
fn main() {
    let n = 10000;
    let searches = 1000;
    let arr: Vec<i32> = (0..n).collect();

    let mut total = 0;
    for i in 0..searches {
        let target = i % n;
        let mut left = 0;
        let mut right = n - 1;

        while left <= right {
            let mid = (left + right) / 2;
            if arr[mid as usize] == target {
                total += mid;
                break;
            } else if arr[mid as usize] < target {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
    }

    std::process::exit(0);
}
